helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -795,6 +798,54 @@ private RebalanceType getRebalanceType(ResourceControllerDataProvider cache,
     }
   }
 
+  private RebalanceType getRebalanceTypePerMessage(Map<String, Integer> requiredStates, Message message,
```

Comment:
This method has not been used. Right? Is it because it is skeleton PR?

Reply:
Not used. I was trying to make each PR independent and easy to review. Then we can focus on the logic of the content itself.